src/mixins/itext_click_behavior.mixin.ts

```diff
@@ -128,8 +123,8 @@ export abstract class ITextClickBehaviorMixin<EventSpec extends ObjectEvents>
    * initializing a mousedDown on a text area will cancel fabricjs knowledge of
    * current compositionMode. It will be set to false.
    */
-  _mouseDownHandler({ e }: TransformEvent) {
-    if (!this.canvas || !this.editable || (e.button && e.button !== 1)) {
+  _mouseDownHandler({ e, button }: TPointerEventInfo) {
+    if (!this.canvas || !this.editable || isRightClick(button)) {
```

Comment:
i see 2 things here ( and on `_mouseDownHandlerBefore` ):
- we move to evaluate the button option rather than the event.button.
- we changed from a `not a left click` to a `is a right click` that leaves the middle click and other possible clicks in an new state ( i would definitely avoid that ).

On top of that it looks like we expose on the event this button value that is not the real button value from the event but more a fabricjs custom value. I wonder if we should remove that value completely and let people look at the event itself

Reply:
ok seems like we used to look at button because `buttons` wasn't on IE. But that is a topic or another day.
I think the only important thing here is to change this isRightClick with a `!isLeftClick` as it was before so we don't change behaviour

Reply:
I didn't mean to change behavior. It was unclear to me

Reply:
`(e.button && e.button !== 1)` means to me:
value is not 0 (not left click) and not middle click => is right click
Why middle click is alright I didn't understand.

Reply:
I think it is best you do it

Reply:
I just don't want to chane the behavioru in a typing PR.
I think the reason of being `not a left click` is because by default fabric wasn't doing anything else with other clicks ( other than allowing for event firing ). So as long as it matters to us, right, middle and the other 2 clicks ( called 4 and 5 that are in some custom mouse ) we just don't do anythin.